test/preview/components/rankers/test_similarity.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+from haystack.preview import Document, ComponentError
+from haystack.preview.components.rankers.similarity import SimilarityRanker
+
+
+class TestSimilarityRanker:
+    @pytest.mark.integration
+    def test_to_dict(self):
+        component = SimilarityRanker(model_name_or_path="cross-encoder/ms-marco-MiniLM-L-6-v2")
+        data = component.to_dict()
+        assert data == {
+            "type": "SimilarityRanker",
+            "init_parameters": {"device": "cpu", "model_name_or_path": "cross-encoder/ms-marco-MiniLM-L-6-v2"},
+        }
+
+    @pytest.mark.integration
+    def test_to_dict_with_custom_init_parameters(self):
+        component = SimilarityRanker()
+        data = component.to_dict()
+        assert data == {
+            "type": "SimilarityRanker",
+            "init_parameters": {"device": "cpu", "model_name_or_path": "cross-encoder/ms-marco-MiniLM-L-6-v2"},
+        }
+
+    @pytest.mark.integration
+    def test_from_dict(self):
+        data = {
+            "type": "SimilarityRanker",
+            "init_parameters": {"device": "cpu", "model_name_or_path": "cross-encoder/ms-marco-MiniLM-L-6-v2"},
+        }
+        component = SimilarityRanker.from_dict(data)
+        assert component.model_name_or_path == "cross-encoder/ms-marco-MiniLM-L-6-v2"
+
+    @pytest.mark.integration
+    def test_run(self):
+        """
+        Test if the component runs correctly.
+        """
+        sampler = SimilarityRanker(model_name_or_path="cross-encoder/ms-marco-MiniLM-L-6-v2")
+        sampler.warm_up()
+        docs = [Document(text="Berlin"), Document(text="Belgrade"), Document(text="Sarajevo")]
+        query = "City in Bosnia and Herzegovina"
+        output = sampler.run(query=query, documents=docs)
+        docs = output["documents"]
+        assert len(docs) == 3
+        assert docs[0].text == "Sarajevo"
+
+        # another test to make sure the first one was not a fluke
```

Comment:
:eyes: This is strange! We froze the random seeds, results should be deterministic. Is this test flaky?

Reply:
No, no it's ok, I just wanted to try two examples to make sure one test is not a correct by fluke (say position of correct document that is first by ranking)